src/connectors/query-rules/__tests__/connectQueryRules-test.ts

```diff
@@ -0,0 +1,568 @@
+
+      const {
+        userData,
+        instantSearchInstance,
+        widgetParams,
+      } = renderingParameters;
+
+      expect(userData).toEqual([]);
+      expect(instantSearchInstance).toEqual({});
+      expect(widgetParams).toEqual({});
+    });
+
+    test('calls the render function on render', () => {
+      const helper = createFakeHelper();
+      const widget = makeWidget({});
+
+      widget.init({ helper, state: helper.state, instantSearchInstance: {} });
+      widget.render({
+        helper,
+        results: new SearchResults(helper.getState(), [{ hits: [] }]),
+        instantSearchInstance: {},
+      });
+
+      {
+        const [renderingParameters, isFirstRender] = renderFn.mock.calls[1];
+
+        expect(isFirstRender).toBe(false);
+
+        const {
+          userData,
+          instantSearchInstance,
+          widgetParams,
+        } = renderingParameters;
+
+        expect(userData).toEqual([]);
+        expect(instantSearchInstance).toEqual({});
+        expect(widgetParams).toEqual({});
+      }
+
+      widget.render({
+        helper,
+        results: new SearchResults(helper.getState(), [
+          { hits: [], userData: [{ banner: 'image.png' }] },
+        ]),
+        instantSearchInstance: {},
+      });
+
+      {
+        const [renderingParameters, isFirstRender] = renderFn.mock.calls[2];
+
+        expect(isFirstRender).toBe(false);
+
+        const {
+          userData,
+          instantSearchInstance,
+          widgetParams,
+        } = renderingParameters;
+
+        expect(userData).toEqual([{ banner: 'image.png' }]);
+        expect(instantSearchInstance).toEqual({});
+        expect(widgetParams).toEqual({});
+      }
+    });
+
+    test('calls the unmount function on dispose', () => {
+      const helper = createFakeHelper();
+      const widget = makeWidget({});
+
+      widget.init({ helper, state: helper.state, instantSearchInstance: {} });
+      widget.dispose({ state: helper.getState() });
+
+      expect(unmountFn).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('options', () => {
+    describe('transformItems', () => {
+      test('is applied to userData', () => {
+        const helper = createFakeHelper();
+        const widget = makeWidget({
+          transformItems: items => items[0],
+        });
+
+        widget.init({ helper, state: helper.state, instantSearchInstance: {} });
+        widget.render({
+          helper,
+          results: new SearchResults(helper.getState(), [
+            {
+              hits: [],
+              userData: [{ banner: 'image1.png' }, { banner: 'image2.png' }],
+            },
+          ]),
+          instantSearchInstance: {},
+        });
+
+        const [renderingParameters] = renderFn.mock.calls[1];
+
+        const { userData } = renderingParameters;
+
+        expect(userData).toEqual({ banner: 'image1.png' });
+      });
+    });
+
+    //     describe('trackedFilters', () => {
```

Comment:
We dropped the implementation we can drop the tests too. Save them on a different branch to avoid to loose them, but no need to push them immediately.